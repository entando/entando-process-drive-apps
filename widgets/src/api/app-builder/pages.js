import { METHODS, DOMAINS } from 'api/constants';
import WIDGETS_CONFIG from 'mocks/app-builder/widgets';
import WIDGET_CONFIG_MOCKUPS from 'mocks/app-builder/pages';
import makeRequest from 'api/makeRequest';

// used to fetch mock response using frameId, it depends on WIDGETS_CONFIG setup
const getWidgetConfigMockResponse = frameId => {
  const widgetType = Object.keys(WIDGETS_CONFIG).find(
    iteratedType => WIDGETS_CONFIG[iteratedType].frameId === frameId
  );
  return WIDGET_CONFIG_MOCKUPS[widgetType];
};

export const getPageWidget = (pageCode, frameId) =>
  makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: `/api/pages/${pageCode}/widgets/${frameId}`,
    method: METHODS.GET,
    mockResponse: getWidgetConfigMockResponse(frameId) || {},
    useAuthentication: true,
    // forceMock: true,
  });

export const putPageWidget = (pageCode, frameId, configs) =>
  makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: `/api/pages/${pageCode}/widgets/${frameId}`,
    method: METHODS.PUT,
    body: configs,
    useAuthentication: true,
  });

// TODO: for ease of development - remove when authetication token is managed by wrapper
export const authenticate = async () => {
  const formData = {
    username: 'admin',
    password: 'adminadmin',
    grant_type: 'password',
  };
  return makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: '/api/oauth/token',
    method: METHODS.POST,
    mockResponse: {
      access_token: 'MOCKED_ACCESS_TOKEN',
      token_type: 'bearer',
      refresh_token: 'MOCKED_REFRESH_TOKEN',
      expires_in: 3599,
    },
    body: Object.keys(formData)
      .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(formData[key])}`)
      .join('&'),
    useAuthentication: false,
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      Authorization: `Basic ${btoa('appbuilder:appbuilder_secret')}`,
    },
  });
};
