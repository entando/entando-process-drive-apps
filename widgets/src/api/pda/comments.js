import { METHODS, DOMAINS } from '../constants';

import makeRequest from '../makeRequest';

import MOCKED_GET_TASK_COMMENTS_RESPONSE from '../../mocks/taskComments/getComments';
import MOCKED_POST_COMMENT_RESPONSE from '../../mocks/taskComments/postComment';
import MOCKED_DELETE_COMMENT_RESPONSE from '../../mocks/taskComments/deleteComment';

export const getTaskComments = async (connection, taskId) => {
  return makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}/comments`,
    method: METHODS.GET,
    mockResponse: MOCKED_GET_TASK_COMMENTS_RESPONSE,
    useAuthentication: true,
  });
};

export const postTaskComment = async (connection, taskId, comment) => {
  return makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}/comments`,
    method: METHODS.POST,
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ comment }),
    mockResponse: MOCKED_POST_COMMENT_RESPONSE,
    useAuthentication: true,
  });
};

export const deleteTaskComment = async (connection, taskId, commentId) => {
  return makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}/comments/${commentId}`,
    method: METHODS.DELETE,
    mockResponse: MOCKED_DELETE_COMMENT_RESPONSE,
    useAuthentication: true,
  });
};
