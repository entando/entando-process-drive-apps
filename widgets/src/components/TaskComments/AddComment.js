import React from 'react';
import PropTypes from 'prop-types';
import i18next from 'i18next';
import withStyles from '@material-ui/core/styles/withStyles';
import { TextareaAutosize, Button, CircularProgress, Typography } from '@material-ui/core';

const MAX_CHARS = 200;

const styles = {
  addNoteField: {
    width: '100%',
    border: '1px solid #E5E6E7',
  },
  addNoteLabel: {
    color: '#A1A1A1',
    margin: '6px 2px',
  },
  addNoteLimit: {
    fontSize: '10px',
    margin: '2px',
  },
  actionButtons: {
    marginTop: '25px',
    display: 'flex',
    justifyContent: 'flex-end',
  },
};

class AddComment extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      comment: '',
    };

    this.onChangeComment = this.onChangeComment.bind(this);
    this.handleAddCommentClick = this.handleAddCommentClick.bind(this);
  }

  onChangeComment({ target: { value } }) {
    this.setState({ comment: value });
  }

  handleAddCommentClick() {
    const { comment } = this.state;
    if (comment.trim()) {
      const { onClickAddComment } = this.props;

      onClickAddComment(comment.trim());
    }
    this.setState({ comment: '' });
  }

  render() {
    const { comment } = this.state;
    const { classes, loading } = this.props;
    const remainingChars = MAX_CHARS - comment.length;

    return (
      <div>
        <label htmlFor="comment-entry" className={classes.addNoteLabel}>
          {i18next.t('taskComments.addNote')}
        </label>
        <TextareaAutosize
          id="comment-entry"
          className={classes.addNoteField}
          rowsMin={2}
          value={comment}
          onChange={this.onChangeComment}
          disabled={loading}
        />
        <Typography
          color={remainingChars < 0 ? 'error' : 'initial'}
          className={classes.addNoteLimit}
        >
          {i18next.t('taskComments.maxChars', { max: MAX_CHARS, rem: remainingChars })}
        </Typography>
        <div className={classes.actionButtons}>
          {loading ? (
            <Button disabled variant="outlined" color="primary">
              <CircularProgress size="20px" />
            </Button>
          ) : (
            <Button onClick={this.handleAddCommentClick} variant="outlined" color="primary">
              {i18next.t('common.add')}
            </Button>
          )}
        </div>
      </div>
    );
  }
}

AddComment.propTypes = {
  classes: PropTypes.shape({
    addNoteField: PropTypes.string,
    addNoteLabel: PropTypes.string,
    addNoteLimit: PropTypes.string,
    actionButtons: PropTypes.string,
  }).isRequired,
  onClickAddComment: PropTypes.func,
  loading: PropTypes.bool,
};

AddComment.defaultProps = {
  onClickAddComment: () => {},
  loading: false,
};

export default withStyles(styles)(AddComment);
