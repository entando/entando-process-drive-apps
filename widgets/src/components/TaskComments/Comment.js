import React from 'react';
import PropTypes from 'prop-types';
import i18next from 'i18next';

import Avatar from '@material-ui/core/Avatar';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = {
  comment: {
    display: 'flex',
    padding: '5px 10px',
    margin: '10px 0px',
  },
  avatar: {
    margin: '0px 30px',
  },
  commentText: {
    fontSize: '13px',
    lineHeight: '15px',
    color: '#686C6E',
  },
  created: {
    fontSize: '10px',
    lineHeight: '12px',
    color: '#8B8888',
    marginTop: '10px',
    marginLeft: '7px',
  },
  removeAnswers: {
    marginLeft: '5px',
    marginRight: '5px',
  },
};

class Comment extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      prompt: false,
    };

    this.togglePrompt = this.togglePrompt.bind(this);
    this.handleClickRemove = this.handleClickRemove.bind(this);
  }

  togglePrompt() {
    const { prompt } = this.state;
    this.setState({ prompt: !prompt });
  }

  handleClickRemove() {
    const {
      comment: { id },
      onClickRemoveComment,
    } = this.props;

    onClickRemoveComment(id);
  }

  render() {
    const { prompt } = this.state;
    const {
      classes,
      comment: { text, createdBy, createdAt },
    } = this.props;

    return (
      <div className={classes.comment}>
        <div className={classes.avatar}>
          <Avatar alt={createdBy} src="/" />
        </div>
        <div className={classes.commentText}>
          <div>{text}</div>
          <div className={classes.created}>
            {`${createdBy} - ${new Date(createdAt).toLocaleString(i18next.language || 'en')} - `}
            <span
              onClick={this.togglePrompt}
              onKeyPress={this.togglePrompt}
              role="button"
              tabIndex={0}
            >
              {i18next.t('taskComments.remove')}
            </span>
            {prompt && (
              <span>
                <span
                  onClick={this.handleClickRemove}
                  onKeyPress={this.handleClickRemove}
                  role="button"
                  tabIndex={0}
                  className={classes.removeAnswers}
                >
                  {i18next.t('common.yes')}
                </span>
                /
                <span
                  onClick={this.togglePrompt}
                  onKeyPress={this.togglePrompt}
                  role="button"
                  tabIndex={0}
                  className={classes.removeAnswers}
                >
                  {i18next.t('common.no')}
                </span>
              </span>
            )}
          </div>
        </div>
      </div>
    );
  }
}

Comment.propTypes = {
  classes: PropTypes.shape({
    comment: PropTypes.string,
    avatar: PropTypes.string,
    commentText: PropTypes.string,
    created: PropTypes.string,
    removeAnswers: PropTypes.string,
  }).isRequired,
  comment: PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    createdBy: PropTypes.string,
    createdAt: PropTypes.string,
  }).isRequired,
  onClickRemoveComment: PropTypes.func,
};

Comment.defaultProps = {
  onClickRemoveComment: () => {},
};

export default withStyles(styles)(Comment);
