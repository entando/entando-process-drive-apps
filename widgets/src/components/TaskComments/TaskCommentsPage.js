import React from 'react';
import PropTypes from 'prop-types';

const TaskCommentsPage = ({ frameId, pageCode, taskId, serviceUrl }) => {
  return (
    <task-comments
      task-id={taskId}
      page-code={pageCode}
      frame-id={frameId}
      service-url={serviceUrl}
    />
  );
};

TaskCommentsPage.propTypes = {
  frameId: PropTypes.string,
  pageCode: PropTypes.string,
  taskId: PropTypes.string.isRequired,
  serviceUrl: PropTypes.string,
};

TaskCommentsPage.defaultProps = {
  frameId: '',
  pageCode: '',
  serviceUrl: '',
};

export default TaskCommentsPage;
