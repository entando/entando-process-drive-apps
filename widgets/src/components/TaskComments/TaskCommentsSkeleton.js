import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import Skeleton from '@material-ui/lab/Skeleton';
import Divider from '@material-ui/core/Divider';

const styles = {
  divider: {
    marginTop: '15px',
    marginBottom: '20px',
  },
  comment: {
    display: 'flex',
    marginTop: '10px',
    marginBottom: '5px',
    paddingLeft: '10px',
  },
  avatar: {
    margin: '5px 25px 5px 30px',
  },
  commentText: {
    marginTop: '5px',
  },
  commentInfo: {
    marginLeft: '15px',
    marginTop: '10px',
  },
  addComment: {
    marginTop: '30px',
  },
  textField: {
    marginTop: '40px',
  },
  controls: {
    display: 'flex',
    justifyContent: 'flex-end',
    marginTop: '20px',
  },
};

const GeneralInformationSkeleton = ({ classes }) => (
  <>
    <div>
      <Skeleton width="190px" height={20} variant="rect" />
      <Divider className={classes.divider} />
    </div>
    <div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="90px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="220px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="320px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.addComment}>
        <Skeleton width="50px" variant="rect" />
        <Divider className={classes.textField} />
      </div>
      <div className={classes.controls}>
        <Skeleton width="70px" height={30} variant="rect" />
      </div>
    </div>
  </>
);

GeneralInformationSkeleton.propTypes = {
  classes: PropTypes.shape({
    divider: PropTypes.string,
    comment: PropTypes.string,
    avatar: PropTypes.string,
    commentText: PropTypes.string,
    commentInfo: PropTypes.string,
    addComment: PropTypes.string,
    textField: PropTypes.string,
    controls: PropTypes.string,
  }).isRequired,
};

export default withStyles(styles)(GeneralInformationSkeleton);
