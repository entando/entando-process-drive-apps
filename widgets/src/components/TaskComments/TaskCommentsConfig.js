/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';
import { FormGroup, ControlLabel, HelpBlock, Row, Col } from 'patternfly-react';

import { getConnections } from 'api/pda/connections';
import Notification from 'components/common/Notification';

import 'patternfly-react/dist/css/patternfly-react.css';
import 'patternfly/dist/css/patternfly.css';
import 'patternfly/dist/css/patternfly-additions.css';

class TaskCommentsConfig extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      sourceList: [],
      errorMessage: '',
      config: {
        knowledgeSource: '',
      },
    };

    this.onChangeKnowledgeSource = this.onChangeKnowledgeSource.bind(this);
    this.closeNotification = this.closeNotification.bind(this);
    this.handleError = this.handleError.bind(this);
    this.fetchScreen = this.fetchScreen.bind(this);
  }

  async componentDidMount() {
    try {
      // getting list of Kie server connections
      const sourceList = await getConnections();
      this.setState({ sourceList: sourceList.payload }, this.fetchScreen);
    } catch (error) {
      this.handleError(error.message);
    }
  }

  componentDidUpdate(prevProps) {
    const { config } = this.props;

    // refetch state if config changes
    if (JSON.stringify(config) !== JSON.stringify(prevProps.config)) {
      this.fetchScreen();
    }
  }

  onChangeKnowledgeSource(e) {
    const { config } = this.state;
    const knowledgeSource = e.target ? e.target.value : e;
    this.setState({ config: { ...config, knowledgeSource } });
  }

  closeNotification = () => {
    this.setState({ errorMessage: '' });
  };

  handleError(errorMessage) {
    this.setState({ errorMessage });
  }

  fetchScreen() {
    const { config } = this.props;
    if (config && config.knowledgeSource) {
      this.onChangeKnowledgeSource(config.knowledgeSource);
    }
  }

  render() {
    const { sourceList, errorMessage, config } = this.state;
    const { knowledgeSource } = config;

    return (
      <div>
        <Notification type="error" message={errorMessage} onClose={this.closeNotification} />
        <form>
          <Row>
            <Col xs={12}>
              <FormGroup controlId="connection">
                <ControlLabel>Knowledge Source</ControlLabel>
                <select
                  className="form-control"
                  value={knowledgeSource}
                  onChange={this.onChangeKnowledgeSource}
                >
                  <option value="">Select...</option>
                  {sourceList.map(source => (
                    <option key={source.name} value={source.name}>
                      {source.name}
                    </option>
                  ))}
                </select>
                <HelpBlock>Select one of the Kie server connections.</HelpBlock>
              </FormGroup>
            </Col>
          </Row>
        </form>
      </div>
    );
  }
}

TaskCommentsConfig.propTypes = {
  config: PropTypes.shape({
    knowledgeSource: PropTypes.string,
  }).isRequired,
};

export default TaskCommentsConfig;
