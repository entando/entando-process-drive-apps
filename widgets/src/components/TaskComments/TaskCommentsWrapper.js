import withAuth from 'components/common/authentication/withAuth';
import TaskComments from 'components/TaskComments/TaskComments';

export default withAuth(TaskComments, [
  'task-comments-create',
  'task-comments-delete',
  'task-comments-list',
]);
