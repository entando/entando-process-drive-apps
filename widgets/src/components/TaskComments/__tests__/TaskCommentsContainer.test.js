import { render, wait } from '@testing-library/react';
import React from 'react';
import 'mocks/i18nMock';

import WIDGET_CONFIGS from 'mocks/app-builder/widgets';
import MOCKED_GET_TASK_COMMENTS_RESPONSE from 'mocks/taskComments/getComments';
import { getTaskComments } from 'api/pda/comments';
import { getPageWidget } from 'api/app-builder/pages';

import TaskComments from 'components/TaskComments/TaskComments';

jest.mock('api/app-builder/pages');
jest.mock('api/pda/comments');

beforeEach(() => {
  getPageWidget.mockClear();
  getTaskComments.mockClear();
});

describe('<TaskComments />', () => {
  getPageWidget.mockImplementation(() => Promise.resolve(WIDGET_CONFIGS.TASK_COMMENTS.configs));
  getTaskComments.mockImplementation(() =>
    Promise.resolve(MOCKED_GET_TASK_COMMENTS_RESPONSE.WITH_COMMENTS)
  );

  const fixedDate = new Date(2020, 0, 1);
  global.Date = jest.fn(() => fixedDate);
  global.Date.now = jest.fn(() => fixedDate);

  it('renders snapshot correctly', async () => {
    const { container } = render(
      <TaskComments
        taskId={WIDGET_CONFIGS.TASK_COMMENTS.taskId}
        pageCode={WIDGET_CONFIGS.TASK_COMMENTS.pageCode}
        frameId={WIDGET_CONFIGS.TASK_COMMENTS.frameId}
        widgetCode={WIDGET_CONFIGS.TASK_COMMENTS.widgetCode}
      />
    );

    await wait(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
