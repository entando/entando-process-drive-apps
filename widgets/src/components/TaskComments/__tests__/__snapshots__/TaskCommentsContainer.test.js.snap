// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskCommentsContainer /> renders snapshot correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root WidgetBox-widgetBox-224 MuiPaper-outlined"
    >
      <div
        class="MuiBox-root MuiBox-root-254"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          taskComments.title
           - 
          7@forms-sample_1.0.1-SNAPSHOT
        </h3>
        <div>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiBox-root MuiBox-root-313"
      >
        <div
          class="TaskComments-commentContainer-24"
        >
          <div
            class="Comment-comment-314"
          >
            <div
              class="Comment-avatar-315"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="pamAdmin"
                  class="MuiAvatar-img"
                  src="/"
                />
              </div>
            </div>
            <div
              class="Comment-commentText-316"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis volutpat est ipsum, quis congue est pharetra ut. Morbi mollis sem eu tellus elementum, et bibendum dolor viverra. Donec maximus quam ligula, et commodo risus ultricies vel. Morbi mollis dui odio, id accumsan turpis pulvinar vitae. Mauris fringilla in mauris nec ullamcorper. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed eu sagittis metus, vitae posuere neque. Etiam ultrices, leo a convallis pharetra, magna ante mollis est, ut tristique neque mauris quis lacus.

In pulvinar, leo quis congue molestie, metus dolor sagittis magna, nec efficitur leo tortor at orci. Sed fringilla urna non urna dapibus, ac imperdiet libero condimentum. Phasellus vulputate vel est eu facilisis. Pellentesque eu purus egestas, condimentum ante finibus, consectetur purus. Sed commodo ante aliquam, vestibulum est at, laoreet lorem. Etiam blandit mi sed lectus fermentum, sit amet lacinia ligula egestas. Curabitur id elit mattis, ultricies mauris sagittis, viverra nulla. Nunc id ligula a dolor placerat porttitor ut in neque. Curabitur imperdiet felis a magna fringilla, nec tempus justo scelerisque.
              </div>
              <div
                class="Comment-created-317"
              >
                pamAdmin - 1/1/2020, 12:00:00 AM - 
                <span
                  role="button"
                  tabindex="0"
                >
                  taskComments.remove
                </span>
              </div>
            </div>
          </div>
          <div
            class="Comment-comment-314"
          >
            <div
              class="Comment-avatar-315"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="pamAdmin"
                  class="MuiAvatar-img"
                  src="/"
                />
              </div>
            </div>
            <div
              class="Comment-commentText-316"
            >
              <div>
                Aenean felis nisi, elementum eu pretium sit amet, condimentum vitae ligula. Quisque facilisis lorem sit amet diam gravida luctus non sit amet eros. Donec id pulvinar lectus. Nunc at sollicitudin nunc. In lorem urna, molestie at ex auctor, tincidunt congue mi. Quisque nunc eros, efficitur eu ex dictum, tincidunt convallis velit. Phasellus et dui metus. Pellentesque ut varius augue, nec maximus elit. Mauris aliquam turpis vel enim egestas rhoncus. Aliquam facilisis magna id nibh cursus auctor. Mauris tellus mauris, auctor eu elementum id, maximus in lorem. Aenean in justo lacinia, malesuada nibh vitae, eleifend nisi. Nulla dapibus odio et dui ullamcorper congue. Sed porttitor sit amet leo et suscipit. Ut ac mattis metus, id venenatis ipsum. In quam metus, semper a magna ac, laoreet eleifend ex.
              </div>
              <div
                class="Comment-created-317"
              >
                pamAdmin - 1/1/2020, 12:00:00 AM - 
                <span
                  role="button"
                  tabindex="0"
                >
                  taskComments.remove
                </span>
              </div>
            </div>
          </div>
          <div
            class="Comment-comment-314"
          >
            <div
              class="Comment-avatar-315"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="pamAdmin"
                  class="MuiAvatar-img"
                  src="/"
                />
              </div>
            </div>
            <div
              class="Comment-commentText-316"
            >
              <div>
                Pellentesque pretium bibendum eros eget hendrerit. Aliquam mollis augue nibh, id lobortis.
              </div>
              <div
                class="Comment-created-317"
              >
                pamAdmin - 1/1/2020, 12:00:00 AM - 
                <span
                  role="button"
                  tabindex="0"
                >
                  taskComments.remove
                </span>
              </div>
            </div>
          </div>
        </div>
        <div>
          <label
            class="AddComment-addNoteLabel-327"
            for="comment-entry"
          >
            taskComments.addNote
          </label>
          <textarea
            class="AddComment-addNoteField-326"
            id="comment-entry"
            rows="2"
            style="height: -4px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="AddComment-addNoteField-326"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0);"
            tabindex="-1"
          />
          <p
            class="MuiTypography-root AddComment-addNoteLimit-328 MuiTypography-body1"
          >
            taskComments.maxChars
          </p>
          <div
            class="AddComment-actionButtons-329"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                common.add
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  
</div>
`;
